### ISSUE TITLE
Glacier/Session.h not built when needed

### ISSUE THREAD

**Issue by bernardnormier** [zeroc-ice/ice#1808]
When building Ice C++ with gcc in parallel, I regularly get the following error, which suggests a dependency bug or some other bug in our Makefiles.

Retrying fixes the issue.

```
Compiling [arm64-shared] src/Glacier2Lib/generated/Router.cpp
In file included from src/Glacier2Lib/SessionHelper.cpp:5:
include/Glacier2/SessionHelper.h:16:10: fatal error: Glacier2/Session.h: No such file or directory
   16 | #include <Glacier2/Session.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:27: src/Glacier2Lib/build/aarch64-linux-gnu/shared/pic/SessionHelper.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from src/Glacier2Lib/generated/Router.cpp:20:
include/generated/Glacier2/Router.h:22:10: fatal error: Glacier2/Session.h: No such file or directory
   22 | #include <Glacier2/Session.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:27: src/Glacier2Lib/build/aarch64-linux-gnu/shared/pic/Router.o] Error 1
```

**Comment by bernardnormier** [comment]
Probably a duplicate of #2124.
